modules/search-service-impl/src/main/java/org/opencastproject/search/impl/solr/SolrRequester.java

```diff
@@ -631,8 +634,14 @@ private SolrQuery getForAction(SearchQuery q, String action, boolean applyPermis
       sb.append(q.getQuery());
     }
 
+    User user = securityService.getUser();
     String solrIdRequest = StringUtils.trimToNull(q.getId());
     if (solrIdRequest != null) {
+
+      // If user has ROLE_EPISODE_<ID>_READ/WRITE, no further permission checks are necessary
+      if (user.hasRole(ACL_ID_PREFIX_EPISODE + solrIdRequest + action.toUpperCase())) {
```

Comment:
Oof.  Subtle bug is subtle.  And annoying.

```suggestion
      if (user.hasRole(ACL_ID_PREFIX_EPISODE + solrIdRequest + "_" + action.toUpperCase())) {
```